File: test/TestProject1/Test1.cs
namespace TestProject1
{
    [TestClass]
    public class Test1 : PageTest
    {
        [AssemblyInitialize]
        public static void AssemblyInit(TestContext context)
        {
            // This method is called once for the test assembly, before any tests are run.
        }

        [AssemblyCleanup]
        public static void AssemblyCleanup()
        {
            // This method is called once for the test assembly, after all tests are run.
        }

        [ClassInitialize]
        public static void ClassInit(TestContext context)
        {
            // This method is called once for the test class, before any tests of the class are run.
        }

        [ClassCleanup]
        public static void ClassCleanup()
        {
            // This method is called once for the test class, after all tests of the class are run.
        }

        [TestInitialize]
        public void TestInit()
        {
            // This method is called before each test method.
        }

        [TestCleanup]
        public void TestCleanup()
        {
            // This method is called after each test method.
        }

        [TestMethod]
        public async Task HomepageHasPlaywrightInTitleAndGetStartedLinkLinkingToTheIntroPage()
        {
            await Page.GotoAsync("https://playwright.dev");

            // Expect a title "to contain" a substring.
            await Expect(Page).ToHaveTitleAsync(new Regex("Playwright"));

            // create a locator
            var getStarted = Page.Locator("text=Get Started");

            // Expect an attribute "to be strictly equal" to the value.
            await Expect(getStarted).ToHaveAttributeAsync("href", "/docs/intro");

            // Click the get started link.
            await getStarted.ClickAsync();

            // Expects the URL to contain intro.
            await Expect(Page).ToHaveURLAsync(new Regex(".*intro"));
        }
    }
}

File: template_feed/Microsoft.DotNet.Test.ProjectTemplates.9.0/content/MSTest-VisualBasic-TestClass/Test1.vb
Imports Microsoft.VisualStudio.TestTools.UnitTesting

Namespace Company.TestProject1
    <TestClass>
    Public Class Test1
#if (Fixture == AssemblyInitialize)
        <AssemblyInitialize>
        Public Shared Sub AssemblyInit(ByVal testContext As TestContext)
            ' This method is called once for the test assembly, before any tests are run.
        End Sub

#endif
#if (Fixture == AssemblyCleanup)
        <AssemblyCleanup>
        Public Shared Sub AssemblyCleanup()
            ' This method is called once for the test assembly, after all tests are run.
        End Sub

#endif
#if (Fixture == ClassInitialize)
        <ClassInitialize>
        Public Shared Sub ClassInit(ByVal testContext As TestContext)
            ' This method is called once for the test class, before any tests of the class are run.
        End Sub

#endif
#if (Fixture == ClassCleanup)
        <ClassCleanup>
        Public Shared Sub ClassCleanup()
            ' This method is called once for the test class, after all tests of the class are run.
        End Sub

#endif
#if (Fixture == TestInitialize)
        <TestInitialize>
        Public Sub TestInit()
            ' This method is called before each test method.
        End Sub

#endif
#if (Fixture == TestCleanup)
        <TestCleanup>
        Public Sub TestCleanup()
            ' This method is called after each test method.
        End Sub

#endif
        <TestMethod>
        Sub TestSub()

        End Sub
    End Class
End Namespace

